p2p/peer_tracker_test.go

```diff
@@ -8,47 +8,77 @@ import (
 
 	"github.com/ipfs/go-datastore"
 	"github.com/ipfs/go-datastore/sync"
-	"github.com/libp2p/go-libp2p/core/peer"
 	"github.com/libp2p/go-libp2p/p2p/net/conngater"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/go-header/p2p/peerstore"
 )
 
 func TestPeerTracker_GC(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer t.Cleanup(cancel)
+
+	gcCycleDefault = time.Millisecond * 200
+	maxAwaitingTime = time.Millisecond
+
 	h := createMocknet(t, 1)
-	gcCycle = time.Millisecond * 200
 	connGater, err := conngater.NewBasicConnectionGater(sync.MutexWrap(datastore.NewMapDatastore()))
 	require.NoError(t, err)
-	p := newPeerTracker(h[0], connGater)
-	maxAwaitingTime = time.Millisecond
-	pid1 := peer.ID("peer1")
-	pid2 := peer.ID("peer2")
-	pid3 := peer.ID("peer3")
-	pid4 := peer.ID("peer4")
+
+	mockPeerStore := peerstore.NewPeerStore(sync.MutexWrap(datastore.NewMapDatastore()))
+	p := NewPeerTracker(h[0], connGater, mockPeerStore)
+
+	peerlist, err := peerstore.GenerateRandomPeerlist(4)
+	require.NoError(t, err)
+
+	pid1 := peerlist[0].ID
+	pid2 := peerlist[1].ID
+	pid3 := peerlist[2].ID
+	pid4 := peerlist[3].ID
+
+	// Add peer with low score to test if it will be GC'ed (it should)
 	p.trackedPeers[pid1] = &peerStat{peerID: pid1, peerScore: 0.5}
+	// Add peer with high score to test if it won't be GCed (it shouldn't)
 	p.trackedPeers[pid2] = &peerStat{peerID: pid2, peerScore: 10}
+
+	// Add peer such that their prune deadlnie is in the past (after GC cycle time has passed)
+	// to test if they will be prned (they should)
 	p.disconnectedPeers[pid3] = &peerStat{peerID: pid3, pruneDeadline: time.Now()}
-	p.disconnectedPeers[pid4] = &peerStat{peerID: pid4, pruneDeadline: time.Now().Add(time.Minute * 10)}
-	assert.True(t, len(p.trackedPeers) > 0)
-	assert.True(t, len(p.disconnectedPeers) > 0)
+	// Add peer such that their prune deadline is not the past (after GC cycle time has passed)
+	// to test if they won't be pruned (they shouldn't)
+	p.disconnectedPeers[pid4] = &peerStat{peerID: pid4, pruneDeadline: time.Now().Add(time.Millisecond * 300)}
 
 	go p.track()
 	go p.gc()
 
-	time.Sleep(time.Second * 1)
+	<-time.After(gcCycleDefault + time.Millisecond*20)
+
+	p.peerLk.Lock()
+	assert.True(t, len(p.trackedPeers) > 0)
+	assert.True(t, len(p.disconnectedPeers) > 0)
+	p.peerLk.Unlock()
```

Comment:
Can we instead introduce a getter in API for such fields? It's super ugly for tests to get such deep.

Reply:
done